packages/vaex-arrow/vaex_arrow/dataset.py

```diff
@@ -0,0 +1,81 @@
+__author__ = 'maartenbreddels'
+import logging
+
+import numpy as np
+import pyarrow as pa
+
+import vaex.dataset
+import vaex.file.other
+logger = logging.getLogger("vaex_arrow")
+
+
+class DatasetArrow(vaex.dataset.DatasetLocal):
+    """Implements storage using arrow"""
+
+    def __init__(self, filename=None, table=None, write=False):
+        super(DatasetArrow, self).__init__(name=filename, path=filename, column_names=[])
+        self._write = write
+        if table is None:
+            self._load()
+        else:
+            self._load_table(table)
+
+    def _load(self):
+        source = pa.memory_map(self.path)
+        reader = pa.open_stream(source)
+        table = pa.Table.from_batches([b for b in reader])
+        self._load_table(table)
+    
+    def _load_table(self, table):
+        self._length_unfiltered =  self._length_original = table.num_rows
+        for col in table.columns:
```

Comment:
This function converts arrow arrays to numpy masked arrays or does it more?

Reply:
Yes, basically does that. I need to refactor it a bit, move some of the code to convert.py. But again, it would be nice of pyarrow does all of this.